src/components/Barrier.jsx

```diff
@@ -53,19 +56,22 @@ export default connect(
         color: store.color,
         hidePriceLines: store.hidePriceLines,
         lineStyle: store.lineStyle,
+        isInitialized: store.isInitialized,
     }),
     (store, {
         color, shadeColor, shade, high, low, relative, draggable, onChange, hidePriceLines, lineStyle,
     }) => {
-        if (color) { store.color = color; }
-        if (shadeColor) { store.shadeColor = shadeColor; }
-        if (shade) { store.shadeState = `SHADE_${shade}`.toUpperCase(); }
-        if (high !== undefined) { store.high_barrier = high; }
-        if (low !== undefined) { store.low_barrier = low; }
-        if (relative !== undefined) { store.relative = relative; }
-        if (draggable !== undefined) { store.draggable = draggable; }
-        if (onChange) { store.onBarrierChange = onChange; }
-        store.lineStyle = lineStyle;
-        store.hidePriceLines = !!hidePriceLines;
+        store.initializePromise.then(action(() => {
```

Comment:
this solves the race condition issue in binary-static; no props will be set, nor barrier will be rendered, until the chart has data.